DataFormats/SiPixelClusterSoA/interface/alpaka/SiPixelClustersDevice.h

```diff
@@ -0,0 +1,55 @@
+#ifndef DataFormats_SiPixelClusterSoA_SiPixelClustersDevice_h
+#define DataFormats_SiPixelClusterSoA_SiPixelClustersDevice_h
+
+#include <cstdint>
+#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "DataFormats/SiPixelClusterSoA/interface/SiPixelClustersHost.h"
+#include "DataFormats/Portable/interface/alpaka/PortableCollection.h"
+#include "DataFormats/SiPixelClusterSoA/interface/SiPixelClustersLayout.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToHost.h"
+
+// TODO: The class is created via inheritance of the PortableCollection.
+// This is generally discouraged, and should be done via composition.
+// See: https://github.com/cms-sw/cmssw/pull/40465#discussion_r1067364306
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class SiPixelClustersDevice : public PortableCollection<SiPixelClustersLayout<>> {
```

Comment:
The problem here is that `alpaka_serial_sync::SiPixelClustersDevice` is a different type than the`SiPixelClustersHost` (I'll clarify this requirement in the [README](https://github.com/cms-sw/cmssw/blob/master/HeterogeneousCore/AlpakaCore/README.md#data-formats)).

The `PortableCollection` itself is a backend-dependent type alias to `PortableCollectionHost` and `PortableCollectionDevice`
https://github.com/cms-sw/cmssw/blob/c4038b4c091680c64a4b4b8876084996ac59d0a8/DataFormats/Portable/interface/alpaka/PortableCollection.h#L17-L30

A possible fix could be
```cpp
#ifdef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
  using SiPixelClustersDevice = SiPixelClustersHost;
#else
  class SiPixelClustersDevice : public PortableCollection<SiPixelClustersLayout<>> {
  ...
#endif
```
(although if we continue this with `#ifdef`, I think we should consider defining our own for "any host backend". macro)

An alternative could be
```cpp
  class SiPixelClustersDeviceImpl : public PortableCollection<SiPixelClustersLayout<>> { ... };
  using SiPixelClustersDevice = std::conditional_t<std::is_same_v<Platform, alpaka::PltfCpu>, SiPixelClustersHost, SiPixelClustersDeviceImpl>;
```
although in this case I think we should consider abstracting the `std::conditional_t<std::is_same_v<Platform, alpaka::PltfCpu>, HostType, DeviceType>` part (since the use case seems to be more widespread than `PortableCollection`).